lmdeploy/serve/async_engine.py

```diff
@@ -475,52 +471,46 @@ def stream_infer(
         # set random if it is not set
         if gen_config.random_seed is None:
             gen_config.random_seed = random.getrandbits(64)
-        prompt_num = len(prompts)
         outputs = Queue()
         generators = []
-        for j in range(0, prompt_num, self.instance_num):
-            batch_prompts = prompts[j:j + self.instance_num]
-            generators = []
-            for i, prompt in enumerate(batch_prompts):
-                generators.append(
-                    self.generate(prompt,
-                                  i,
-                                  gen_config=gen_config,
-                                  stream_response=True,
-                                  sequence_start=True,
-                                  sequence_end=True,
-                                  do_preprocess=do_preprocess,
-                                  adapter_name=adapter_name,
-                                  **kwargs))
-
-            async def _inner_call(i, generator):
-                async for out in generator:
-                    outputs.put(
-                        Response(out.response, out.generate_token_len,
-                                 out.input_token_len, i + j, out.finish_reason,
-                                 out.token_ids, out.logprobs))
-
-            async def gather():
-                await asyncio.gather(*[
-                    _inner_call(i, generators[i])
-                    for i in range(len(batch_prompts))
-                ])
-                outputs.put(None)
-
-            proc = Thread(
-                target=lambda: _get_event_loop().run_until_complete(gather()))
-            proc.start()
-
-            while True:
-                try:
-                    out = outputs.get(timeout=0.001)
-                    if out is None:
-                        break
-                    yield out
-                except Empty:
-                    pass
-
-            proc.join()
+        for i, prompt in enumerate(prompts):
```

Comment:
On what occasion batch streaming infer is needed?

Reply:
It was just a demand from community.